zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -1365,23 +1234,39 @@ impl<F: FieldExt> MPTConfig<F> {
                     offset += 1;
                 }
 
-                if self.check_zeros {
-                    for ind in 0..(33 * 255) {
-                        region.assign_fixed(
-                            || "fixed table",
-                            self.fixed_table[0],
-                            offset,
-                            || Value::known(F::from(FixedTableTag::RangeKeyLen256 as u64)),
-                        )?;
+                let max_length = 34i32 + 1;
+                for (tag, range) in [
+                    (FixedTableTag::RangeKeyLen256, 256),
+                    (FixedTableTag::RangeKeyLen16, 16),
+                ] {
+                    for n in /* -192-max_length */ -512..max_length {
+                        let range = if n < 0 { 1 } else { range };
+                        for idx in 0..range {
```

Comment:
I think it would be good to have a short description of this table, something like: "These fixed rows enable to easily check whether there are zeros in the unused columns (the number of unused columns vary). The last two columns contain the rows `(0, -512), (0, -511), ..., (0, -1), (0, 0), (1, 0), ..., (255, 0), (1, 0), (1, 1), ..., (1, 255), (2, 0), (2, 1), ...`. The lookups ensure that when the unused columns start, the values in these columns are zeros - when the unused columns start, the value that is used for the lookup in the last column is negative and thus a zero is enforced on the value that is used in the first column."